contrib/cryptomb/private_key_providers/source/cryptomb_private_key_provider.cc

```diff
@@ -0,0 +1,793 @@
+#include "contrib/cryptomb/private_key_providers/source/cryptomb_private_key_provider.h"
+
+#include <memory>
+
+#include "envoy/registry/registry.h"
+#include "envoy/server/transport_socket_config.h"
+
+#include "source/common/config/datasource.h"
+
+#include "openssl/ec.h"
+#include "openssl/ssl.h"
+
+// BoringSSL internal definitions, needed for calculating ECDSA ephemeral key.
+// TODO(ipuustin): ask BoringSSL maintainers to expose these in the BoringSSL headers?
```

Comment:
@agl @davidben Hi Adam and David! Do you think it would be possible to expose `ec_random_nonzero_scalar()`, `ec_scalar_to_bytes()` and the related `EC_SCALAR` data type in BoringSSL public API? They are required if you want to calculate ECDSA ephemeral keys the way BoringSSL does it.

Reply:
I don't think we're likely to do that because we keep eyeing whether we can put a version-skew coupling around the FIPS module.

(And it goes without saying that copy-pasting bits of internal headers is invalid and this change cannot land while doing that.)

Reply:
In this case, do you think there could be some other way how BoringSSL could provide access to this functionality? It seems to me that the only non-trivial function needed for this is `bn_rand_range_words()` and the functions it calls. I would love to use ready-made functions rather than implementing it myself. OpenSSL has library access to ECDSA ephemeral key generation, and I think I can't be the only person needing this.